pandas/tests/series/test_analytics.py

```diff
@@ -1015,6 +1015,12 @@ def test_unique(self):
         tm.assert_categorical_equal(s.unique(), pd.Categorical([np.nan]),
                                     check_dtype=False)
 
+    def test_unique_obj_none_preservation(self):
+        # GH 20866
+        s = pd.Series(['foo', None])
+        result = s.unique()
+        assert result[1] is None
```

Comment:
construct array and compare. this should be in test_algorithms

Reply:
Make sure to use `strict_nan=True` in that case

Reply:
good point